src/pnpq/devices/waveplate_stub.py

```diff
@@ -0,0 +1,235 @@
+        # Simulated delay for homing
+        # TODO: Delay calculation
+        time.sleep(1)
+        self.logger.info(f"Home Position: {self.current_position}")
+
+    def auto_update_start(self):
+        self.__ensure_port_open()
+        self.logger.info("Stub Waveplate Auto Update Start")
+        # TODO: Auto updating logging
+
+    def auto_update_stop(self):
+        self.__ensure_port_open()
+        self.logger.info("Stub Waveplate Auto Update Stop")
+        # TODO: Auto updating logging
+
+    def disable_channel(self, chanid: int):
+        self.__ensure_port_open()
+        self.logger.info(f"Stub Waveplate Disable Channel {chanid}")
+
+        if chanid >= self.max_channel:
+            raise WaveplateInvalidMotorChannelError(f"Channel {chanid} is not enabled")
+
+        # Remove channel from enabled channels
+        self.enabled_channels.remove(chanid)
+
+        # Simulated delay
+        time.sleep(0.1)
+
+    def enable_channel(self, chanid: int):
+        self.__ensure_port_open()
+        self.logger.info(f"Stub Waveplate Enable Channel {chanid}")
+        if chanid > self.max_channel:
+            raise WaveplateInvalidMotorChannelError(f"Invalid motor channel: {chanid}. Max channel: {self.max_channel}")
+
+        self.enabled_channels.append(chanid)
+
+        # Simulated delay
+        time.sleep(0.1)
+
+        # TODO: Return a fake reply from the device
+
+    def device_resolution(self) -> int:
+        return self.resolution
+
+    def getpos(self):
+        self.__ensure_port_open()
+        self.logger.info("Stub Waveplate Get Position")
+        self.logger.info(f"Current Position: Steps: {self.current_position} Degrees: {self.current_position / self.resolution}")
+        return self.current_position
+
+    def rotate(self, degree: int | float):
+        self.__ensure_port_open()
+        self.__ensure_valid_degree(degree)
+
+        if not self.__stub_check_channel(1):
+            # Do nothing if channel is not enabled
+            return
+
+        self.logger.info(f"Stub Waveplate Rotate to {degree}")
+        # Calculate number of steps to move
+        movePosition = degree * self.resolution
+        # Update current position
+        self.current_position = movePosition
+        # Delay to simulate rotation (for now: v=1ms/deg)
+        time.sleep(abs(movePosition - self.current_position) / 1000)
+        # TODO: Return a fake reply from the device
+
+    def step_backward(self, steps: int):
+        self.__ensure_port_open()
+        self.__ensure_less_than_max_steps(steps)
+
+        self.logger.info(f"Stub Waveplate Step Backware {steps}")
+
+        if not self.__stub_check_channel(1):
+            # Do nothing if channel is not enabled
+            return
+
+        # Make steps negative
+        steps = -steps
+        self.current_position += steps
+        # Delay to simulate rotation (for now: v=1ms/deg)
+        time.sleep(abs(steps) / 1000)
+
+    def step_forward(self, steps: int):
```

Comment:
I think that `step_forward` and `step_backward` can be implemented simply by calling `rotate` instead of copy/pasting `rotate`'s code?

Reply:
The real device also seems to re-implement the rotating action. But it seems unnecessary, so let's not do that, and maybe we should fix the real one too.

Reply:
I remembered why this is done like this. There seems to be an inconsistency between the functions. Some takes a degree value before converting to steps, and others uses steps directly. I wonder if this inconsistency is there on purpose. I'll do a manual conversion for now.